.github/workflows/iroha2-dev.yml

```diff
@@ -19,75 +21,67 @@ jobs:
       - uses: actions/checkout@v2
       - uses: Swatinem/rust-cache@v1
       - name: set up buildx
-        uses: docker/setup-buildx-action@v1
+        uses: docker/setup-buildx-action@v2
         id: buildx
         with:
           install: true
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build and push docker image
-        uses: docker/build-push-action@v2
+          driver: docker
+
+      - name: Build iroha2:base image
+        uses: docker/build-push-action@v3
         with:
+          load: true
+          tags: iroha2:base
           context: .
-          push: true
-          tags: hyperledger/iroha2:dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-      - name: Build and push docker image (iroha_client_cli)
-        uses: docker/build-push-action@v2
+          file: Dockerfile.base
+
+      - name: Build iroha2:build docker image
+        uses: docker/build-push-action@v3
         with:
+          load: true
+          tags: iroha2:build
           context: .
-          push: true
-          tags: hyperledger/iroha2:client-cli-dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-            BIN=iroha_client_cli
-      - name: Build and push docker image (kagami)
-        uses: docker/build-push-action@v2
+          file: Dockerfile.build
+
+      - name: Build iroha2:dev docker image
+        run: |
+          docker build -f Dockerfile -t hyperledger/iroha2:dev .
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
         with:
-          context: .
-          push: true
-          tags: hyperledger/iroha2:kagami-dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-            BIN=kagami
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Push iroha2:dev docker image
+        run: |
+          docker push hyperledger/iroha2:dev
 
       - name: Build and push docker image (load-rs:dev)
         run: |
           sleep 10s
           echo "wait to other workflow"
-      - uses: convictional/trigger-workflow-and-wait@v1.6.0
+      - uses: convictional/trigger-workflow-and-wait@v1.6.2
         with:
           owner: soramitsu
           repo: iroha2-longevity-load-rs
           github_token: ${{ secrets.G_ACCESS_TOKEN }}
           workflow_file_name: load-rs-push-from-dev.yaml
           ref: dev
 
-  build-ci-image:
+  upload_coverage:
+    if: github.event.pull_request.merged
```

Comment:
![image](https://user-images.githubusercontent.com/49983831/179160481-26a98f30-a510-4c13-b695-cc61dd6a5469.png)

This no reaction of workflows can imply some syntax errors. Could you try this:

```suggestion
    if: github.event.pull_request.merged == true
```

Also, this line should be inserted to other jobs in this workflow

I'll introduce [`actionlint`](https://github.com/rhysd/actionlint#actionlint) in my next PR. Even with `actionlint`, I don't know why the current expression can be invalid though

Reply:
Probably I was wrong with the cause of the no reaction. Maybe it is the conflicts

![image](https://user-images.githubusercontent.com/49983831/179179211-da76ef3e-99df-4e4d-bf83-1e11eb989cce.png)

Reply:
I don't think that there are syntax errors due all checks for this PR are running on fork.